flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/AdaptiveGraphManager.java

```diff
@@ -291,6 +299,8 @@ private List<JobVertex> createJobVerticesAndUpdateGraph(List<StreamNode> streamN
 
         generateConfigForJobVertices(jobVertexBuildContext);
 
+        onStreamGraphUpdated(streamGraph);
```

Comment:
I think calling this method here is strange because what we want is to update the JSON due to a newly created job vertex.

I suggest that we call generateStreamGraphJson directly here.

Reply:
I added a private function `generateStreamGraphJson()` to `AdaptiveGraphManager` and called it in `onStreamGraphUpdated` & here to make it clearer.